File: backend/user-service/src/main/java/com/whatsyouretf/userservice/domain/auth/dto/AuthResponse.java
package com.whatsyouretf.userservice.domain.auth.dto;

import com.whatsyouretf.userservice.domain.user.entity.User;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
@AllArgsConstructor
public class AuthResponse {

    private String accessToken;
    private String refreshToken;
    private Long expiresIn;
    private Boolean isNewUser;
    private UserInfo user;

    @Getter
    @Builder
    @AllArgsConstructor
    public static class UserInfo {
        private Long id;
        private String email;
        private String nickname;
        private String profileImage;
        private String loginProvider;

        public static UserInfo from(User user, String provider) {
            return UserInfo.builder()
                    .id(user.getId())
                    .email(user.getEmail())
                    .nickname(user.getNickname())
                    .profileImage(user.getProfileImage())
                    .loginProvider(provider)
                    .build();
        }
    }

    public static AuthResponse of(String accessToken, String refreshToken, Long expiresIn, boolean isNewUser, User user, String provider) {
        return AuthResponse.builder()
                .accessToken(accessToken)
                .refreshToken(refreshToken)
                .expiresIn(expiresIn)
                .isNewUser(isNewUser)
                .user(UserInfo.from(user, provider))
                .build();
    }

    // 토큰 갱신용 (refreshToken 없이)
    public static AuthResponse ofRefresh(String accessToken, Long expiresIn) {
        return AuthResponse.builder()
                .accessToken(accessToken)
                .expiresIn(expiresIn)
                .build();
    }
}

File: frontend/WYE/app/src/main/java/com/d102/wye/data/remote/dto/response/AuthResponse.kt
package com.d102.wye.data.remote.dto.response

import com.google.gson.annotations.SerializedName

data class TokenResponse(
    @SerializedName("accessToken")
    val accessToken: String,

    @SerializedName("refreshToken")
    val refreshToken: String,

    @SerializedName("expiresIn")
    val expiresIn: Int,             // 토큰 만료 시간 (초)

    @SerializedName("isNewUser")
    val isNewUser: Boolean,

    @SerializedName("user")
    val user: UserResponse
)

data class UserResponse(
    @SerializedName("id")
    val id: Long,

    @SerializedName("email")
    val email: String?,

    @SerializedName("nickname")
    val nickname: String,

    @SerializedName("profileImage")
    val profileImage: String?,      // nullable (미설정 시 null)

    @SerializedName("loginProvider")
    val loginProvider: String       // "EMAIL" | "KAKAO"
)
